x-pack/plugins/security_solution/server/lib/siem_migrations/rules/api/util/installation.ts

```diff
@@ -31,105 +30,100 @@ const installPrebuiltRules = async (
+    .flatMap((rule) => rule.elastic_rule?.prebuilt_rule_id ?? [])
+    .filter((value, index, self) => self.indexOf(value) === index);
+  const prebuiltRules = await getPrebuiltRules(rulesClient, savedObjectsClient, prebuiltRulesIds);
 
-    // If prebuilt rule is not installed, then keep reference to install it
-    const installableRule = installableRules.find(
-      (rule) => rule.rule_id === ruleToInstall.elastic_rule?.prebuilt_rule_id
-    );
-    if (installableRule) {
-      assetsToInstall.push(installableRule);
+  const { installed: alreadyInstalledRules, installable } = prebuiltRules.reduce(
+    (acc, item) => {
+      if (item.current) {
+        acc.installed.push(item.current);
+      } else {
+        acc.installable.push(item.target);
+      }
+      return acc;
+    },
+    { installed: [], installable: [] } as {
+      installed: RuleResponse[];
+      installable: RuleResponse[];
     }
-  });
-
-  // Filter out any duplicates which can occur when multiple translated rules matched the same prebuilt rule
-  const filteredAssetsToInstall = assetsToInstall.filter(
-    (value, index, self) => index === self.findIndex((rule) => rule.rule_id === value.rule_id)
   );
 
+  // Install prebuilt rules
   // TODO: we need to do an error handling which can happen during the rule installation
-  const { results: installedRules } = await createPrebuiltRules(
+  const { results: newlyInstalledRules } = await createPrebuiltRules(
     detectionRulesClient,
-    filteredAssetsToInstall
+    installable
   );
   await performTimelinesInstallation(securitySolutionContext);
 
+  const installedRules = [
+    ...alreadyInstalledRules,
+    ...newlyInstalledRules.map((value) => value.result),
+  ];
+
+  // Create migration rules updates templates
+  const rulesToUpdate: UpdateRuleMigrationInput[] = [];
   installedRules.forEach((installedRule) => {
-    const rules = rulesToInstall.filter(
-      (rule) => rule.elastic_rule?.prebuilt_rule_id === installedRule.result.rule_id
+    const filteredRules = rulesToInstall.filter(
+      (rule) => rule.elastic_rule?.prebuilt_rule_id === installedRule.rule_id
     );
-    rules.forEach((prebuiltRule) => {
-      rulesToUpdate.push({
-        id: prebuiltRule.id,
+    rulesToUpdate.push(
+      ...filteredRules.map(({ id }) => ({
+        id,
         elastic_rule: {
-          id: installedRule.result.id,
+          id: installedRule.id,
         },
-      });
-    });
+      }))
+    );
   });
 
   return rulesToUpdate;
 };
 
-const installCustomRules = async (
+export const installCustomRules = async (
   rulesToInstall: StoredRuleMigration[],
   detectionRulesClient: IDetectionRulesClient,
   logger: Logger
 ): Promise<UpdateRuleMigrationInput[]> => {
   const rulesToUpdate: UpdateRuleMigrationInput[] = [];
-  await Promise.all(
-    rulesToInstall.map(async (rule) => {
+  const createCustomRulesOutcome = await initPromisePool({
+    concurrency: MAX_RULES_TO_UPDATE_IN_PARALLEL,
```

Comment:
Can we use a specific constant for installation instead of using the one for update?

Reply:
done